### PR TITLE
PLAT-10600 enforce pylint checks in PR builder

### PR DESCRIPTION
**Pull request by symphony-mariacristina**
### Ticket
[PLAT-10600](https://perzoinc.atlassian.net/browse/PLAT-10600)

### Description
Goal of this PR is to enable pylint checks in the PR builder.
For now the limit to fail is set to 9.50 (max being 10.00). We can start like this and if needed we can adapt it later on, if we think that is too strict.

All changes about fixture are related to this issue: https://stackoverflow.com/questions/46089480/pytest-fixtures-redefining-name-from-outer-scope-pylint

### Checklist
- [x] Referenced a ticket in the PR title and in the corresponding section
- [x] Filled properly the description and dependencies, if any
- [x] Unit tests updated or added
- [x] Docstrings added or updated
- [x] Updated the documentation in [docs folder](../docs)

### CODE REVIEW

File: .github/workflows/pylint.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Install Pylint
+        run: pip install pylint
+
+      - name: Download previous run results
```
Comment (symphony-mariacristina) [inline .github/workflows/pylint.yaml:26]: Every run uploads the scan results that can be downloaded in the next run to have a comparison within runs (taking the last successful one).
Generated a message like this at the end of the scan:

```
------------------------------------------------------------------
Your code has been rated at 9.58/10 (previous run: 9.74/10, -0.16)
```

**Comment by thibauult** [comment]
> For now the limit to fail is set to 9.50 

😍

File: .github/workflows/pylint.yaml (hunk 1 above)
Comment (symphony-youri) [reply .github/workflows/pylint.yaml:26]: is it going to comparer the results of the last run for the same branch/PR or for any of the PRs? (the latter would not make much sense right?)

Comment (symphony-elias) [reply .github/workflows/pylint.yaml:26]: Yes we have to download .pylint.d of the latest commit on 2.0 branch

File: .pylintrc [hunk 2]
```diff
@@ -478,7 +479,9 @@ variable-naming-style=snake_case
 
 # This flag controls whether inconsistent-quotes generates a warning when the
 # character used as a quote delimiter is used inconsistently within a module.
-check-quote-consistency=no
+check-quote-consistency=yes
```
Comment (symphony-elias) [inline .pylintrc:482]: 👍

File: .github/workflows/pylint.yaml (hunk 1 above)
Comment (symphony-mariacristina) [reply .github/workflows/pylint.yaml:26]: yes thank you for noticing, I completely missed that. I will add modify it in the next commit

**Review by thibauult — APPROVED** [review]

**Review by symphony-elias — APPROVED** [review]
LGTM

**Review by symphony-youri — APPROVED** [review]

**Pull request merged**